client/me/purchases/confirm-cancel-domain/loading-placeholder.jsx

```diff
@@ -40,4 +41,4 @@ ConfirmCancelDomainLoadingPlaceholder.propTypes = {
 	selectedSite: PropTypes.oneOfType( [ PropTypes.bool, PropTypes.object ] ),
 };
 
-export default ConfirmCancelDomainLoadingPlaceholder;
+export default localize( ConfirmCancelDomainLoadingPlaceholder );
```

Comment:
I'm curious: what does `localize()` do here? I thought that it just added a `translate` prop, but clearly there must be some other reason it's used. Does `useTranslate` have the same effect?

Reply:
> I'm curious: what does localize() do here? I thought that it just added a translate prop, but clearly there must be some other reason it's used. 
Yes, it enhances the component with the i18n props, but it also subscribes the component to the i18n's `change` event and [forces the component to re-render](https://github.com/Automattic/wp-calypso/blob/trunk/packages/i18n-calypso/src/localize.js#L30-L38). This event is being fired when [switching the locale](https://github.com/Automattic/wp-calypso/blob/trunk/packages/i18n-calypso/src/i18n.js#L304) or when [adding translations to the existing Tannin locale data](https://github.com/Automattic/wp-calypso/blob/trunk/packages/i18n-calypso/src/i18n.js#L351), as it is in our case with async loaded translation chunks.

> Does useTranslate have the same effect?
Yes, it [subscribes the component to the same event](https://github.com/Automattic/wp-calypso/blob/trunk/packages/i18n-calypso/src/use-translate.js#L16-L20).

Reply:
🙇‍♂️  Thank you for the explanation! Does this mean that it's a good idea to always use either the HOC or the hook on top-level components?

Reply:
> 🙇‍♂️ Thank you for the explanation! Does this mean that it's a good idea to always use either the HOC or the hook on top-level components?

I don't think it matters that much, but using the hook without assigning it to a variable may look a bit strange, imo. Essentially, it should be the same in terms of performance and behaviour.